packages/data-explorer/src/mappers/mapperUtils.ts

```diff
@@ -0,0 +1,130 @@
+// @ts-ignore
+import api from '@molgenis/molgenis-api-client'
+import { MetaDataRefEntity, DataApiResponse, MetaDataAttribute, MetaDataCategoricalOption } from '@/types/ApiResponse'
+import { FilterOption, FilterOptionsPromise } from '@/types/ApplicationState'
+
+/**
+ * Build a function that returns a Promise of an array containing objects of type FieldOption
+ *
+ * Simple types like STRING or TEXT do not have input properties, in this case 'null' is returned
+ * The returned function returns a Promise of an array consisting of type FieldOption
+ *
+ * @example Example schema for generating field options
+ * const schema = {
+    *  fields: [
+    *    {
+    *      id: 'example',
+    *      label: 'Example field',
+    *      options: () => Promise.resolve([
+    *          {
+    *            value: '1',
+    *            text: 'Example option 1'
+    *          },
+    *          {
+    *            value: '2',
+    *            text: 'Example option 2'
+    *          }
+    *        ]
+    *    }
+    *  ]
+    * }
+    *
+    * @param attribute
+    * @param options MapperOptions optional object containing options to configure mapper
+    * @returns {Function|null} Function which returns a Promise representing an Array of FieldOptions
```

Comment:
Suggest to alway return a promise returning function , this makes it easier to handle for caller